code/game/machinery/spaceheater.dm

```diff
@@ -234,6 +235,159 @@
+				beaker.reagents.handle_reactions()
+			if(HEATER_MODE_HEAT)
+				if(targetTemperature < beaker.reagents.chem_temp)
+					return
+				beaker.reagents.adjust_thermal_energy((targetTemperature - beaker.reagents.chem_temp) * power_mod * delta_time * SPECIFIC_HEAT_DEFAULT * beaker.reagents.total_volume)
+			if(HEATER_MODE_COOL)
+				if(targetTemperature > beaker.reagents.chem_temp)
+					return
+				beaker.reagents.adjust_thermal_energy((targetTemperature - beaker.reagents.chem_temp) * power_mod * delta_time * SPECIFIC_HEAT_DEFAULT * beaker.reagents.total_volume)
+		var/requiredEnergy = heatingPower * delta_time * (power_mod * 4)
+		cell.use(requiredEnergy / efficiency)
+		beaker.reagents.handle_reactions()
+	update_icon()
+
+/obj/machinery/space_heater/improvised_chem_heater/ui_data()
+	. = ..()
+	.["chemHacked"] = TRUE
+	.["beaker"] = beaker
+	.["currentTemp"] = beaker ? (round(beaker.reagents.chem_temp - T0C)) : "N/A"
+	return .
+
+/obj/machinery/space_heater/improvised_chem_heater/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("ejectBeaker")
+			//Eject doesn't turn it off, so you can preheat for beaker swapping
+			replace_beaker(usr)
+			. = TRUE
+
+///Slightly modified to ignore the open_hatch - it's always open, we hacked it.
+/obj/machinery/space_heater/improvised_chem_heater/attackby(obj/item/I, mob/user, params)
+	add_fingerprint(user)
+	if(default_unfasten_wrench(user, I))
+		return
+	else if(istype(I, /obj/item/stock_parts/cell))
+		if(cell)
+			to_chat(user, "<span class='warning'>There is already a power cell inside!</span>")
+			return
+		else if(!user.transferItemToLoc(I, src))
+			return
+		cell = I
+		I.add_fingerprint(usr)
+
+		user.visible_message("<span class='notice'>\The [user] inserts a power cell into \the [src].</span>", "<span class='notice'>You insert the power cell into \the [src].</span>")
+		SStgui.update_uis(src)
+	//reagent containers
+	if(istype(I, /obj/item/reagent_containers) && !(I.item_flags & ABSTRACT) && I.is_open_container())
+		. = TRUE //no afterattack
+		var/obj/item/reagent_containers/B = I
+		if(!user.transferItemToLoc(B, src))
+			return
+		replace_beaker(user, B)
+		to_chat(user, "<span class='notice'>You add [B] to [src]'s water bath.</span>")
+		updateUsrDialog()
+		return
+	//Dropper tools
+	if(beaker)
+		if(is_type_in_list(I, list(/obj/item/reagent_containers/dropper, /obj/item/ph_meter, /obj/item/ph_paper, /obj/item/reagent_containers/syringe)))
+			I.afterattack(beaker, user, 1)
+			return
+
+	else if(default_deconstruction_crowbar(I))
+		return
+
+/obj/machinery/space_heater/improvised_chem_heater/on_deconstruction()
+	. = ..()
+	var/bonus_junk = list(
+		/obj/item/stack/cable_coil = 2,
+		/obj/item/stack/sheet/glass = 2,
+		/obj/item/stack/sheet/iron = 2,
+		/obj/item/thermometer = 1
+		)
+	for(var/item in bonus_junk)
+		if(prob(80))
+			new item(get_turf(loc))
+	return
```

Comment:
```suggestion
	var/static/bonus_junk = list(
		/obj/item/stack/cable_coil = 2,
		/obj/item/stack/sheet/glass = 2,
		/obj/item/stack/sheet/iron = 2,
		/obj/item/thermometer = 1
		)
	for(var/item in bonus_junk)
		if(prob(80))
			new item(get_turf(loc))
```
* We can cache the list so it's only created a single time.
* No need to have empty returns at the end of procs.